### PR TITLE
docs: Add documentation on how to add plugins

### PR DESCRIPTION
**Pull request by machadoit**

### CODE REVIEW

**Comment by machadoit** [comment]
@pauloribeiro-codacy do you know if we have good markdown configuration file to follow? So we can also update the README to follow them?

File: README.md [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+    ```typescript
+      console.log("Generate xss description files")
+      await docGenerator.downloadDocs(
+        (pattern) =>
+          `${githubBaseUrl}/Rantanen/eslint-plugin-xss/master/docs/rules/${pattern}.md`,
+        "xss"
+      )
+    ```
+
+1.  Generate documentation so it adds the new plugin documentation:
```
Comment (prcr) [inline README.md]: @machadoit, it seems like this step also only applies when adding plugins. :thinking:

**Comment by prcr** [comment]
Hello @machadoit, I did a few edits to the new section with instructions on how to add plugins to ESLint. Please check the commit message for a list of the changes that I did and feel free to say if you don't agree with any of the changes. :+1: 

Regarding a configuration file for Markdown linting, you can check the one I have on the codacy/docs repository, which is quite simple:

https://github.com/codacy/docs/blob/master/.remarkrc

It basically uses a couple of presets that are very "sane" ([preset-lint-consistent](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-consistent) and [preset-lint-recommended](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-recommended)), and just two more rules to ensure that heading levels increment by one at a time and to define the unordered list marker as the hyphen.

**Comment by prcr** [comment]
OK, now I understand better why you were asking for a Markdown configuration file. Codacy reported lots of issues after my changes. :sweat_smile: In the meanwhile I already fixed most of them, but there's still one issue that I don't understand why it is triggered because the indentation is correct. :confused:

**Comment by machadoit** [comment]
@pauloribeiro-codacy I guess it looks better with the "generate documentation" as step. The motivation was that you may want to regenerate it without adding a plugin, like when updating the tool. What do you think?

Either way, thanks for the changed, LGTM!

**Comment by machadoit** [comment]
Also added the suggested configuration file

**Comment by prcr** [comment]
> @pauloribeiro-codacy I guess it looks better with the "generate documentation" as step. The motivation was that you may want to regenerate it without adding a plugin, like when updating the tool. What do you think?

I makes sense to have it in its own section then - I was thinking that could be the reason why you had already separated those instructions. :smile: :+1: 

I reverted it back.

**Review by prcr — APPROVED** [review]

**Comment by lolgab** [comment]
> Regarding a configuration file for Markdown linting, you can check the one I have on the codacy/docs repository, which is quite simple:
> 
> https://github.com/codacy/docs/blob/master/.remarkrc
> 
> It basically uses a couple of presets that are very "sane" ([preset-lint-consistent](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-consistent) and [preset-lint-recommended](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-recommended)), and just two more rules to ensure that heading levels increment by one at a time and to define the unordered list marker as the hyphen.

Could we use the same "sane" presets as new Codacy defaults? What do you think @pauloribeiro-codacy

**Comment by prcr** [comment]
> > Regarding a configuration file for Markdown linting, you can check the one I have on the codacy/docs repository, which is quite simple:
> > https://github.com/codacy/docs/blob/master/.remarkrc
> > It basically uses a couple of presets that are very "sane" ([preset-lint-consistent](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-consistent) and [preset-lint-recommended](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-preset-lint-recommended)), and just two more rules to ensure that heading levels increment by one at a time and to define the unordered list marker as the hyphen.
> 
> Could we use the same "sane" presets as new Codacy defaults? What do you think @pauloribeiro-codacy

@lolgab Yes, we should definitely think about that. Perhaps we would have to activate the individual patterns that are included in those presets. :thinking: And it could be a good idea to compare what we have now with what the presets suggest, just so we are more aware of the changes.

If it's feasible to do this without a big impact for customers, I can add a task on my side to start preparing the change. 🙂

**Comment by lolgab** [comment]
> If it's feasible to do this without a big impact for customers, I can add a task on my side to start preparing the change. 🙂

It impacts only on newly created projects. So it should be fine to change :-)

**Comment by prcr** [comment]
> > If it's feasible to do this without a big impact for customers, I can add a task on my side to start preparing the change. slightly_smiling_face
> 
> It impacts only on newly created projects. So it should be fine to change :-)

I added https://codacy.atlassian.net/browse/DOCS-110, it will be great to understand if there is an opportunity to improve the Markdown analysis here. :+1:

**Pull request merged**